src/Diagnostics/Debug/StateCheck.jl

```diff
@@ -0,0 +1,556 @@
+        if MPI.Comm_rank(MPI.COMM_WORLD) == 0
+            println(
+                "# SC =======|============|==========|",
+                minStr,
+                "|",
+                maxStr,
+                "|",
+                aveStr,
+                "|",
+                stdStr,
+                "|",
+            )
+        end
+
+        ## Iterate over the set of MPIStateArrays for this callback
+        for f in fields
+            olabel = f[2]
+            olStr = @sprintf("%12.12s", olabel)
+            mArray = f[1]
+
+            # Get descriptor for MPIStateArray
+
+            V = typeof(mArray).parameters[2]
+
+            ## Iterate over fields in each MPIStateArray
+            #  (use ivar to index individual arrays within the MPIStateArray)
+            ivar = 0
+            statsVal_dict = Dict()
+            for i in 1:length(V.names)
+                for n in flattenednames(fieldtype(V, i), prefix = fieldname(V, i))
+                    ivar = ivar + 1
+                    nStr = @sprintf("%9.9s", n)
+                    if MPI.Comm_rank(MPI.COMM_WORLD) == 0
+                        print("# SC ", nSStr, "|", olStr, "|", nStr, " |")
+                    end
+                    statsString = scstats(mArray, ivar, nprec)
+                    if MPI.Comm_rank(MPI.COMM_WORLD) == 0
+                        println(
+                            statsString[1],
+                            "|",
+                            statsString[2],
+                            "|",
+                            statsString[3],
+                            "|",
+                            statsString[4],
+                            "|",
+                        )
+                    end
+                    statsVal_dict[n] = statsString[5]
+                    append!(
+                        curStats_flat,
+                        [[
+                            olabel,
+                            n,
+                            statsString[5].max,
+                            statsString[5].min,
+                            statsString[5].mean,
+                            statsString[5].std,
+                        ]],
+                    )
+                end
+            end
+            curStats_dict[olabel] = statsVal_dict
+        end
+        if MPI.Comm_rank(MPI.COMM_WORLD) == 0
+            println("# SC +++++++++++ClimateMachine StateCheck call-back end+++++++++++++++++++")
+        end
+    end;
+
+    if MPI.Comm_rank(MPI.COMM_WORLD) == 0
+        println("# SC Finish: creating state check callback")
+    end;
+
+    return cb
+)
+
+# Anything else prints doc
+sccreate(Any...) = (println("# ERROR: Function signature not matched");
+@doc sccreate)
+sccreate() = sccreate(0)
+
+function scstats(V, ivar, nprec)
+
+    # Get number of MPI procs
+    nproc = MPI.Comm_size(V.mpicomm)
+
+    npr = nprec
+    fmt = @sprintf("%%%d.%de", npr + 8, npr)
+
+    # Min
+    phiLoc = minimum(V.realdata[:, ivar, :])
```

Comment:
It might be nice to write a convenience function for this, since it digs into the internals of the `realdata` memory layout.